duckduckgo/Android
# Add clear data notification ab test

Task/Issue URL: https://app.asana.com/0/361428290920652/950570149935908
Tech Design URL: N/A

**Description**:
Add clear data notification experiment

**Steps to test this PR**:
**TEST PREPARATION:**
1. In VariantManager, comment out all variants other than "me", "mt", "mf", "mu"
1. Open NotificationScheduler and change `scheduleClearDataNotification(duration.toLong(), TimeUnit.DAYS)` to `scheduleClearDataNotification(30, TimeUnit.SECONDS)`
1. Open Charles and prepare to watch networking traffic

**TEST ONE: Notification launched from active app**
1. Launch the app from a clean installation
1. Ensure the notification fires after ~30 seconds
1. Check that the `notification shown` `mnot_s` pixel fires
1. Tap the notification content and ensure the settings page is opened
1. Check that the `notification launched` `mnot_l` pixel fires
1. Launch the app again and ensure the notification does not show

**TEST TWO: Notification launched from killed app**
1. Launch the app from a clean installation
1. Kill the app
1. Ensure the notification fires after ~30 seconds
1. Check that the notification shown `mnot_s` pixel fires
1. Tap the notification content and ensure the settings page is opened
1. Check that the `notification launched` `mnot_l` pixel fires
1. Launch the app again and ensure the notification does not show

**TEST THREE: Notification dismissed**
1. Launch the app from a clean installation
1. Ensure the notification fires after ~30 seconds
1. Check that the notification shown `mnot_s` pixel fires
1. Swipe the notification away to clear it
1. Check that the `notification cancelled` `mnot_c` pixel fires

**TEST FOUR: Notification not shown if clear data already set**
1. Launch the app from a clean installation
1. Update the auto clear data settings so they are not blank
1. Ensure the notification does not fire
1. Kill the app and relaunch
1. Again, ensure the notification does not fire

**TEST FIVE: Disable/Enable Notifications**
1. Disable app notifications
1. Ensure that `mnot_d` pixel fires on next launch
1. Relaunch the app and ensure the `mnot_d` pixel does not fire a second time
1. Enable app notifications
1. Ensure that `mnot_e` pixel fires on next launch
1. Relaunch the app and ensure the `mnot_e` pixel does not fire a second time

**TEST SIX: Database migration**
1. Install the version of the app in `develop` and launch it
2. Upgrade to this branch
3. Make sure nothing explodes 💥

###### Internal references:
[Software Engineering Expectations](https://app.asana.com/0/59792373528535/199064865822552)
[Technical Design Template](https://app.asana.com/0/59792373528535/184709971311943)
